### ISSUE TITLE
RP 1.0.beta11pre2 incompatibility with current Pale Moon (25.8)

### ISSUE THREAD

**Issue by beerisgood** [RequestPolicyContinued/requestpolicy#739]
1.0.beta11pre2 can't be installed in current Pale Moon (25.8) because the minimum version in install.rdf is 36.0.
If i add a entry for the Pale Moon GUID, (which is {8de7fcbb-c55c-4fbe-bfc5-fc555c87dbc4}) the install works, but i have no RP icon and can't add a icon so its not usable.

RP 1.0.beta10.2 works fine so i hope you fix this and add Pale Moon compatibility.

I also get this error in error console:
Error: invalid object initializer
Source File: chrome://rpcontinued/content/lib/environment.jsm
Line: 508

**Comment by myrdd** [comment]
I'm sorry about that, but right now I'm not going to switch back to the old `minVersion`. I did that step because many ES6 features which are very useful are already supported in Gecko, specifically Firefox ESR.

I would have looked for which ES6 features are already supported in Pale Moon, but I could not find any informative page. To be sure RP works on Pale Moon the code must only use ES5 features. I definitely won't do the step back, I aim to support mainly Firefox and SeaMonkey.

However, maybe the solution is a transpiler. If you want to help, you could try using Babel or Google’s Traceur to translate RequestPolicy's code to use only ES5.

**Comment by beerisgood** [comment]
What for ES6 features did you need? Moonchild can this then include in Pale Moon.

**Comment by myrdd** [comment]
This is what I use and which is not in Gecko 25:

* Gecko 36
  * Symbol.iterator
* Gecko 34
  * shorthand method definitions
  * template strings
  * destructuring: spread operator
* Gecko 33
  * shorthand property names

**Comment by myrdd** [comment]
I've updated the list.

I've tried Babel out of curiosity, but I'm not satisfied. It's very slow (slow XPI generation) and it would be another tool. I prefer not to use Babel.

I've evaluated if I can revert the usages of the above features, and RP works again in Pale Moon. However, it hurts not being able to use features like the new Iteration protocol (generator functions and `Symbol.iterator`). I hope Pale Moon will implement the ES6 features soon, otherwise I will have to skip Pale Moon support.
